media-video/mpv/mpv-0.27.1.ebuild

```diff
@@ -0,0 +1,374 @@
+	egl? ( || ( gbm X wayland ) )
+	gbm? ( drm egl opengl )
+	lcms? ( opengl )
+	luajit? ( lua )
+	opengl? ( || ( aqua egl X raspberry-pi !cli ) )
+	raspberry-pi? ( opengl )
+	test? ( opengl )
+	tools? ( cli )
+	uchardet? ( iconv )
+	v4l? ( || ( alsa oss ) )
+	vaapi? ( || ( gbm X wayland ) )
+	vdpau? ( X )
+	wayland? ( egl )
+	X? ( egl? ( opengl ) )
+	xv? ( X )
+	zsh-completion? ( cli )
+	${PYTHON_REQUIRED_USE}
+"
+
+COMMON_DEPEND="
+	!libav? ( >=media-video/ffmpeg-3.2.2:0=[encode?,threads,vaapi?,vdpau?] )
+	libav? ( >=media-video/libav-12:0=[encode?,threads,vaapi?,vdpau?] )
+	alsa? ( >=media-libs/alsa-lib-1.0.18 )
+	archive? ( >=app-arch/libarchive-3.0.0:= )
+	bluray? ( >=media-libs/libbluray-0.3.0:= )
+	cdda? ( dev-libs/libcdio-paranoia )
+	cuda? ( >=media-video/ffmpeg-3.3:0 )
+	drm? ( x11-libs/libdrm )
+	dvd? (
+		>=media-libs/libdvdnav-4.2.0
+		>=media-libs/libdvdread-4.1.0
+	)
+	egl? ( media-libs/mesa[egl,gbm(-)?,wayland(-)?] )
+	iconv? (
+		virtual/libiconv
+		uchardet? ( app-i18n/uchardet )
+	)
+	jack? ( virtual/jack )
+	javascript? ( >=dev-lang/mujs-1.0.0 )
+	jpeg? ( virtual/jpeg:0 )
+	lcms? ( >=media-libs/lcms-2.6:2 )
+	libass? (
+		>=media-libs/libass-0.12.1:=[fontconfig,harfbuzz]
+		virtual/ttf-fonts
+	)
+	libcaca? ( >=media-libs/libcaca-0.99_beta18 )
+	lua? (
+		!luajit? ( <dev-lang/lua-5.3:= )
+		luajit? ( dev-lang/luajit:2 )
+	)
+	openal? ( >=media-libs/openal-1.13 )
+	pulseaudio? ( media-sound/pulseaudio )
+	raspberry-pi? ( >=media-libs/raspberrypi-userland-0_pre20160305-r1 )
+	rubberband? ( >=media-libs/rubberband-1.8.0 )
+	samba? ( net-fs/samba )
+	sdl? ( media-libs/libsdl2[sound,threads,video] )
+	v4l? ( media-libs/libv4l )
+	vaapi? (
+		!libav? ( >=media-video/ffmpeg-3.3:0 )
+		libav? ( >=media-video/libav-13:0 )
+		x11-libs/libva:=[drm?,X?,wayland?]
+	)
+	vdpau? (
+		!libav? ( >=media-video/ffmpeg-3.3:0 )
+		libav? ( >=media-video/libav-13:0 )
+		x11-libs/libvdpau
+	)
+	wayland? (
+		>=dev-libs/wayland-1.6.0
+		>=x11-libs/libxkbcommon-0.3.0
+	)
+	X? (
+		x11-libs/libX11
+		x11-libs/libXScrnSaver
+		x11-libs/libXext
+		x11-libs/libXinerama
+		x11-libs/libXrandr
+		opengl? (
+			x11-libs/libXdamage
+			virtual/opengl
+		)
+		xv? ( x11-libs/libXv )
+	)
+	zlib? ( sys-libs/zlib )
+"
+DEPEND="${COMMON_DEPEND}
+	${PYTHON_DEPS}
+	dev-python/docutils
+	virtual/pkgconfig
+	doc? ( dev-python/rst2pdf )
+	dvb? ( virtual/linuxtv-dvb-headers )
+	test? ( >=dev-util/cmocka-1.0.0 )
+	v4l? ( virtual/os-headers )
+	zsh-completion? ( dev-lang/perl )
+"
+RDEPEND="${COMMON_DEPEND}
+	cuda? ( x11-drivers/nvidia-drivers[X] )
+	selinux? ( sec-policy/selinux-mplayer )
+	tools? ( ${PYTHON_DEPS} )
+"
+
+PATCHES=(
+	"${FILESDIR}/${PN}-0.19.0-make-ffmpeg-version-check-non-fatal.patch"
+	"${FILESDIR}/${PN}-0.23.0-make-libavdevice-check-accept-libav.patch"
+	"${FILESDIR}/${PN}-0.25.0-fix-float-comparisons-in-tests.patch"
+	"${FILESDIR}/${PN}-0.27.0-add-missing-link-flags-for-rpi.patch"
+)
+
+pkg_setup() {
+	[[ ${MERGE_TYPE} != "binary" ]] && python_setup
```

Comment:
You can move this to `src_configure`, then you don't have to check `MERGE_TYPE`. It's a common-ish practice with python-r1.